spec/models/check_availability_task_spec.rb

```diff
@@ -0,0 +1,64 @@
+describe CheckAvailabilityTask do
+  include ::Spec::Support::TenantIdentity
+
+  let(:source) { Source.create!(:name => "source", :tenant => tenant) }
+  let!(:task) { CheckAvailabilityTask.create!(:name => "task", :tenant => tenant, :source => source, :status => "ok", :state => state, :owner => "William") }
+
+  around do |example|
+    Insights::API::Common::Request.with_request(default_request) { example.call }
+  end
+
+  describe "before_update callback" do
+    context "when task is timed out" do
+      let(:state) { "timedout" }
+
+      it "should raise an exception" do
+        expect { task.update(:state => "completed") }.to raise_exception("Task #{task.id} was marked as timed out")
+      end
+    end
+
+    context "when task is running" do
+      let(:state) { "running" }
+
+      it "should update" do
+        expect(task).to receive(:post_check_availability_task)
+        task.update(:state => "completed")
+        expect(task.state).to eq("completed")
+      end
+    end
+  end
+
+  describe "#timed_out" do
+    before { Timecop.safe_mode = true }
+
+    context "when task is completed" do
+      let(:state) { "completed" }
+
+      it "returns false" do
+        expect(task.timed_out?).to be_falsey
+      end
+
+      it "returns false" do
+        Timecop.travel(Time.current + CheckAvailabilityTask.timeout_interval + 10) do
```

Comment:
@hsong-rh can we change the 10 to a variable called tolerance and use it wherever we want to be +/- the timeout_interval